test/built-ins/Temporal/PlainDate/prototype/toPlainDateTime/argument-string-time-designator-required-for-disambiguation.js

```diff
@@ -16,6 +16,14 @@ const ambiguousStrings = [
   "1130",     //   ditto, including DD that doesn't occur in every month
   "12-14",    // ambiguity between MM-DD and HH-UU
   "202112",   // ambiguity between YYYYMM and HHMMSS
+  // same strings as above but with a calendar annotation, which must not cause
+  // disambiguation in favour of time
+  "2021-12[u-ca=iso8601]",
+  "1214[u-ca=iso8601]",
+  "0229[u-ca=iso8601]",
+  "1130[u-ca=iso8601]",
+  "12-14[u-ca=iso8601]",
+  "202112[u-ca=iso8601]",
```

Comment:
Should "same strings as above" be enforced programatically rather than manually, aligning more closely with adding the designator prefix?
```js
const ambiguousStrings = [
  "2021-12",  // ambiguity between YYYY-MM and HHMM-UU
  "1214",     // ambiguity between MMDD and HHMM
  "0229",     //   ditto, including MMDD that doesn't occur every year
  "1130",     //   ditto, including DD that doesn't occur in every month
  "12-14",    // ambiguity between MM-DD and HH-UU
  "202112",   // ambiguity between YYYYMM and HHMMSS
];
ambiguousStrings.forEach((string) => {
  // The ambiguous strings should be rejected with or without a calendar annotation.
  const stringWithCalendar = string + "[u-ca=iso8601]";
  [string, stringWithCalendar].forEach((ambiguousString) => {
    assert.throws(
      RangeError,
      () => instance.toPlainDateTime(ambiguousString),
      `ambiguous string requires a T prefix: ${ambiguousString}`
    );
  });

  // The same strings should be accepted with a time designator prefix
  // (but not with a space prefix).
  [string, stringWithCalendar].forEach((ambiguousString) => {
    instance.toPlainDateTime("T" + ambiguousString);
    instance.toPlainDateTime("t" + ambiguousString);
    const ambiguousStringWithSpacePrefix = " " + ambiguousString;
    assert.throws(
      RangeError,
      () => instance.toPlainDateTime(ambiguousStringWithSpacePrefix),
      `space is not accepted as a substitute for T prefix: ${ambiguousStringWithSpacePrefix}`
    );
  });
});
```
(and likewise elsewhere)

Reply:
Alternatively, it would seem valuable to [template](https://github.com/tc39/test262/blob/main/CONTRIBUTING.md#procedurally-generated-tests) references to a single collection of ambiguous time strings and a single collection of unambiguous time strings, to avoid drift across test files.

Reply:
That's a valid concern. I haven't used the templating facility because IIUC it's not able to handle things like Temporal.PlainTime.compare() having to do the same string conversion tests twice, once for each argument. So, I usually template an initial collection of files and then apply modifications like compare() by hand.

In any case I think the suggestion of concatenating the calendar annotation instead of listing all the strings a second time is a good one, and goes some way towards mitigating the problem.

As a follow up, what would you think of centralizing the lists of strings into `harness/temporalHelpers.js`?

Reply:
> As a follow up, what would you think of centralizing the lists of strings into `harness/temporalHelpers.js`?

I would be in favor of anything that addresses the duplication-and-drift issue.

Reply:
I've made a `TemporalHelpers.ISO` object which we can gradually move collections of strings into, let me know what you think.